### ISSUE TITLE
lib.lua[273]:Failed to rename buffer

### ISSUE THREAD

**Issue by greister** [kyazdani42/nvim-tree.lua#215]
After pull the new commit.
got the error:lib.lua[273]:Failed to rename buffer.

**Comment by curtainp** [comment]
i have the same problem

**Comment by osa1** [comment]
This is introduced with 905afba. Parent commit of that one works fine (491fd68).

**Comment by kyazdani42** [comment]
Hi, I need more information, I havent encountered that issue.

**Comment by kyazdani42** [comment]
I think this is a duplicate of #213 please refer to the docs in the meanwhile.

**Comment by ray-x** [comment]
It seems that `g:nvim_tree_bindings` is no longer needed.  Remove this setting and should works.

**Comment by kyazdani42** [comment]
nvim_tree_bindings is still required if you wish to add new bindings to the tree buffer or you want to override existing ones.

**Comment by koopa1338** [comment]
Its pretty verbose to override keybindings, it would be much easier if the plugin would export the the `get_lua_cb` function, maybe with a better name. This is how remapping looks before and after:

 ```diff
--- old.lua
+++ new.lua
vim.g.nvim_tree_bindings = {
-    cd = {'.'},
-    rename = {'c'},
-    copy = {'y'},
-    prev_git_item = {'gn'},
-    next_git_item = {'gp'},
}

vim.g.nvim_tree_bindings = {
+    ["."] = ":lua require'nvim-tree'.on_keypress('cd')<CR>",
+    ["c"] = ":lua require'nvim-tree'.on_keypress('rename')<CR>",
+    ["y"] = ":lua require'nvim-tree'.on_keypress('copy')<CR>",
+    ["gn"] = ":lua require'nvim-tree'.on_keypress('next_git_item')<CR>",
+    ["gp"] = ":lua require'nvim-tree'.on_keypress('prev_git_item')<CR>",
}

```

**Comment by kyazdani42** [comment]
you are right, i'll push something today to fix these issues.

**Comment by kyazdani42** [comment]
fixed in 58e44d29
